src/app/convai-checker.component.spec.ts

```diff
@@ -1858,4 +1909,186 @@ describe('Convai checker test', () => {
     // Send an input event to trigger the service call.
     setTextAndFireInputEvent(queryTexts[callCount], textArea);
   }));
+
+  it('Test gradient colors', async(() => {
```

Comment:
Another test I personally think would be better were it broken up into smaller units saying specifically what it is you're trying to test.  Same note as above (note that this has a lot of separate sections with comments denoting what is being tested, those would be excellent parts of this to just turn into separate test cases)

Reply:
I see your point, but I think there is also value in a larger test that checks how the component handles changes to these settings as they occur (which is actually where a lot of bugs end up popping up) rather than the first time the settings are set. That's why a lot of these tests are very long, because they're really testing sequences of actions. And the longer tests do take into account everything the shorter tests would, and then some. Happy to chat more about this.